rts/lra/lra-filters/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -214,51 +185,32 @@ public void filter(ContainerRequestContext containerRequestContext) throws IOExc
             return;
         }
 
-        enlist = true;
-        nested = resourceInfo.getResourceMethod().isAnnotationPresent(NestedLRA.class);
-
         switch (type) {
             case MANDATORY: // a txn must be present
                 checkForTx(type, incommingLRA, true);
 
-                if (nested) {
-                    // a new LRA is nested under the incomming LRA
-                    suspendedLRA = incommingLRA;
-                    lraTrace(containerRequestContext, suspendedLRA, "ServerLRAFilter before: MANDATORY start new LRA");
-                    newLRA = lraId = startLRA(incommingLRA, method, getTimeOut(method));
-                } else {
                     lraId = incommingLRA;
                     resumeTransaction(incommingLRA); // txId is not null
-                }
+
                 break;
             case NEVER: // a txn must not be present
                 checkForTx(type, incommingLRA, false);
 
-                if (nested) {
-                    // nested does not make sense
-                    throwGenericLRAException(null, Response.Status.PRECONDITION_FAILED.getStatusCode(),
-                            type.name() + " but found Nested annnotation");
-                }
-
-                enlist = false;
-                lraId = null;
+                lraId = null; // must not run with any context
 
                 break;
             case NOT_SUPPORTED:
-                if (nested) {
-                    // nested does not make sense
-                    throwGenericLRAException(null, Response.Status.PRECONDITION_FAILED.getStatusCode(),
-                            type.name() + " but found Nested annnotation");
-                }
-
-                enlist = false;
                 suspendedLRA = incommingLRA;
-                lraId = null;
+                lraId = null; // must not run with any context
 
                 break;
+            case NESTED:
+                // FALLTHRU
             case REQUIRED:
                 if (incommingLRA != null) {
-                    if (nested) {
+                    if (type == NESTED) {
+                        headers.putSingle(LRA_HTTP_PARENT_CONTEXT_HEADER, incommingLRA.toASCIIString());
```

Comment:
Does anything ever use LRA_HTTP_PARENT_CONTEXT_HEADER - I just see this put and some passing around earlier in the diff?

Reply:
The MP-LRA test suite validates that it is present on participant callbacks and on the business method where the nested LRA is creates

Reply:
OK, thanks

Reply:
There are a few uses but an example of where we validate that it is available on participant callbacks is:  https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java#L277